### DISCUSSION TITLE
Zephyr Vent Hood - Hardware question

### DISCUSSION THREAD

**Discussion by mirthy** [HomeSpan/HomeSpan#736] [General]
Hi!  Just trying to figure out exactly what hardware was used for the Zephyr Vent Hood project.  I have the exact same hood that eats batteries like no tomorrow and the remote just doesn't work a lot of the time like the project says.  If you could, can you share what hardware was used?  I've already put together some of the hardware you mentioned, the most critical piece with the transmitter I need to get the right one.  Is something like the CC1101 or RFM69HCW 433MHz overkill?  Thank you so much for such wonderful projects (HomeSpan and the vent hood project).  I really appreciate all your hard work and dedication!

Here's what I've ordered:
Adafruit HUZZAH32 (PID: 3591)
Adafruit LiPoly 500 mAh battery (PID: 1578)
Momentary push buttons
Breadboard
Jumper wires
Project box

On order still:
RF Link Transmitter - 434MHz - WRL-10534
Solu TI CC1101 Wireless Transceiver module - backup

Tried but unsuccessful:
* el cheapo FS1000A transmitter - OOK keying, but I think the parts are bad because I couldn't get an old Arduino UNO to drive any signal to come out.

**Comment by HomeSpan** [comment]
The hardware I've used is listed on the [Zephyr page](https://github.com/HomeSpan/ZephyrVentHood) (see the links for the transmitter).

This particular project is designed for a simple transmitter and will not work with the RFM (which uses SPI to communicate).

**Reply by mirthy** [reply]
OK, perfect, I just wanted to double check the transmitter.  I'll let you know how my progress goes.  Last question, did you power the transmitter with the USB pin from the HUZZAH32 or another source?

Thanks again!

**Reply by HomeSpan** [reply]
From USB pin. Probably used an apple charger or similar.

**Comment by HomeSpan** [comment]
One thing to note:  To get the Zephyr working I had to first use a receiver to reverse-engineer the codes.  I did this on an actual Arduino plotting the value of a pin and determining if the remote I had was transmitting long or short pulses.  This required a lot of experimentation, so you may want to first see if you can capture the signals for your remote.

**Reply by mirthy** [reply]
Thank you for the experimentation and figuring that part out (it's a little beyond my time and skillset right now)

Some good news is that I can confirm that my vent hood has the same pattern of signals.  A while back, I got a Flipper Zero and I didn't have time to play with it.  I found your project and decided to try getting something working with the vent hood.  I found the Flipper Zero's the Sub-Ghz recording/playback works well to capture and replay the Zephyr's signals.  I'm also able to take the .sub file that gets generated and plot it.  The pattern exactly matches your codes, for example this is the plotted light code for one of the 8 repeated sequences that I captured with the Flipper Zero:
<img width="696" alt="image" src="https://github.com/HomeSpan/HomeSpan/assets/1112440/89cd7a62-7569-4520-9992-44a5cac02443">

I saw another project that can take the raw sub file from the Flipper Zero and send it out using an ESP32 and CC1101 - https://github.com/simondankelmann/Esp32-SubGhz/tree/main/Esp32/Esp32-CC1101-Standalone

I figured your method of transmitting is really straightforward and I'm going to continue that path before I try anything else.  But the Flipper Zero data is a good confirmation that my vent hood is using the same patterns of signals as yours.  I just have to go pickup the right transmitter.  Fortunately, there's a store close by that has it in stock.

**Reply by mirthy** [reply]
Oh yeah, what did you use for an antenna?

**Reply by HomeSpan** [reply]
That's great that your signals match.   Should save a lot of time trying to reverse-engineer.  

For an antenna, just a simple wire about 13 cm or so.  The transmitter should provide you with recommended length.
